samples/bot-feed-members/nodejs/bots/teamsConversationBot.js

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const {
+    ActionTypes,
+    CardFactory,
+    MessageFactory,
+    TeamsActivityHandler,
+    TeamsInfo,
+    TurnContext
+} = require('botbuilder');
+
+// The TeamsConversationBot class extends TeamsActivityHandler to handle Teams-specific activities.
+class TeamsConversationBot extends TeamsActivityHandler {
+    constructor() {
+        super();
+
+        // Handle messages sent to the bot.
+        this.onMessage(async (context, next) => {
+            // Get a paginated list of members in the current Teams conversation.
+            const members = await TeamsInfo.getPagedMembers(context);
+            
+            // Remove the bot's mention from the message text to clean up the user's input.
+            TurnContext.removeRecipientMention(context.activity);
+
+            // Normalize the message text for case-insensitive comparison.
+            const text = context.activity.text.trim().toLocaleLowerCase();
+            
+            // If the message contains 'list', list the members. Otherwise, send a welcome card.
+            if (text.includes('list')) {
+                await this.listMembersAsync(context, members);
+            } else {
+                await this.cardActivityAsync(context, false);
+            }
+            // Call the next middleware in the pipeline.
+            await next();
+        });
+
+        // Handle new members added to the conversation.
+        this.onMembersAdded(async (context, next) => {
+            // Get the ID of the first member added.
+            const memberId = context.activity.membersAdded[0].aadObjectId;
+            
+            // Fetch detailed information about the new member.
+            const member = await TeamsInfo.getMember(context, memberId);
+        
+            // Create an Adaptive Card to welcome the new member and display their details.
+            const memberCard = {
+                type: "AdaptiveCard",
+                body: [
+                    {
+                        type: "TextBlock",
+                        text: `Welcome to the team, ${member.name}!`,
+                        weight: "Bolder",
+                        size: "ExtraLarge"  // Increased size
+                    },
+                    {
+                        type: "TextBlock",
+                        text: "Here are your details:",
+                        weight: "Bolder",
+                        size: "Large",  // Increased size
+                        spacing: "Large"  // Added spacing
+                    },
+                    {
+                        type: "FactSet",
+                        facts: [
+                            { title: "Name:", value: member.name },
```

Comment:
The tenant Id needs to be included in the information about the user, as it is the main difference about federated scenarios. That is the way to know from which tenant each of the members of the group chat are from, and to prove the federated scenario.

Reply:
Updated.

![image](https://github.com/user-attachments/assets/37f232a6-5f77-4a94-8ddb-e1bb17cfd141)

Reply:
Updated